### ISSUE TITLE
DeDRM, python errors on MacOS (Mojave)

### ISSUE THREAD

**Issue by atanasj** [apprenticeharper/DeDRM_tools#1008]
Not sure what is happening here...

Relevant system info: 

```bash
ProductName:	Mac OS X
ProductVersion:	10.14.6
BuildVersion:	18G3020
Python 3.7.6 (user set default)
```

Error output: 

```
calibre, version 4.10.1
ERROR: Unhandled exception: <b>OSError</b>:dlopen(/usr/local/lib/libcrypto.dylib, 6): no suitable image found.  Did find:
	/usr/local/lib/libcrypto.dylib: code signature in (/usr/local/lib/libcrypto.dylib) not valid for use in process using Library Validation: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.

calibre 4.10.1  embedded-python: True is64bit: True
Darwin-18.7.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '18.7.0', 'Darwin Kernel Version 18.7.0: Thu Jan 23 06:52:12 PST 2020; root:xnu-4903.278.25~1/RELEASE_X86_64')
Python 2.7.16
OSX: ('10.14.6', ('', '', ''), 'x86_64')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 7, 0) && Zotero Metadata Importer (1, 0, 63) && Obok DeDRM (6, 7, 0)
Traceback (most recent call last):
  File "site-packages/calibre/gui2/preferences/plugins.py", line 295, in customize_plugin
  File "site-packages/calibre/gui2/preferences/plugins.py", line 373, in modify_plugin
  File "site-packages/calibre/customize/__init__.py", line 157, in do_user_config
  File "calibre_plugins.dedrm.__init__", line 648, in config_widget
  File "site-packages/calibre/customize/zipplugin.py", line 185, in load_module
  File "calibre_plugins.dedrm.config", line 37, in <module>
  File "site-packages/calibre/customize/zipplugin.py", line 185, in load_module
  File "calibre_plugins.dedrm.utilities", line 6, in <module>
  File "site-packages/calibre/customize/zipplugin.py", line 185, in load_module
  File "calibre_plugins.dedrm.ignoblekeygen", line 194, in <module>
  File "calibre_plugins.dedrm.ignoblekeygen", line 188, in _load_crypto
  File "calibre_plugins.dedrm.ignoblekeygen", line 127, in _load_crypto_libcrypto
  File "lib/python2.7/ctypes/__init__.py", line 366, in __init__
OSError: dlopen(/usr/local/lib/libcrypto.dylib, 6): no suitable image found.  Did find:
	/usr/local/lib/libcrypto.dylib: code signature in (/usr/local/lib/libcrypto.dylib) not valid for use in process using Library Validation: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.
```

Please let me know if you need more info. Thanks.

**Comment by ZolaLa9** [comment]
When I get a chance, I will test on Catalina.

In the meantime, I'm not seeing this error on High Sierra running Calibre 3.48 (the last version of Calibre that will run on High Sierra).

**Comment by atanasj** [comment]
Thanks. I just tested again with another document and still not working.

**Comment by uberjay** [comment]
I ran into the same problem (also running mojave -- 10.14.6 (18G3020)).

It looks like the DeDRM plugin ends up trying to load the openssl libcrypto library from /usr/local/lib -- at least for _me_, this copy of libcrypto came along with a Homebrew-installed `mysql`:
```
$ ls  -l /usr/local/lib/libcrypto.*
lrwxr-xr-x 46 xyz 15 Jan 11:49 /usr/local/lib/libcrypto.1.1.dylib -> ../Cellar/mysql/8.0.19/lib/libcrypto.1.1.dylib
lrwxr-xr-x 42 xyz 15 Jan 11:49 /usr/local/lib/libcrypto.dylib -> ../Cellar/mysql/8.0.19/lib/libcrypto.dylib
```
I suspect this is actually a bug in the mysql Homebrew package, since it's known that installing conflicting/newer openssl in /usr/local can be harmful. For example, the `openssl` package is so-called "keg-only" to avoid these problems:
```
openssl@1.1 is keg-only, which means it was not symlinked into /usr/local,
because openssl/libressl is provided by macOS so don't link an incompatible version.
```

TL;DR - `brew uninstall mysql` actually fixed the problem. after restarting calibre I can now strip DRM from epubs successfully!

**Comment by atanasj** [comment]
That works. DRM was stripped from they PDF. Thanks!

**Comment by JoshuaCrewe** [comment]
For the record, I had this issue and uninstalling mysql solved it for me.

I still need mysql though. I found that I can install mariadb and still use this plugin in calibre.

`brew uninstall mysql && brew install mariadb`

👍
